sycl/source/device_selector.cpp

```diff
@@ -137,10 +153,20 @@ select_device(const DSelectorInvocableType &DeviceSelectorInvocable,
 /// 2. CPU
 /// 3. Host
 /// 4. Accelerator
+
+static void traceDeviceSelector(const std::string &DeviceType) {
+  bool ShouldTrace = false;
+  ShouldTrace = detail::pi::trace(detail::pi::TraceLevel::PI_TRACE_BASIC);
+  if (ShouldTrace) {
+    std::cout << "SYCL_PI_TRACE[all]: Requested device_type: " << DeviceType << std::endl;
```

Comment:
In this case it's not aligned with other device selector information that started with `SYCL_PI_TRACE[all]` but it is also printed as basic information in case of successful device initialization. So I did it in the same style for better readability.
```
SYCL_PI_TRACE[all]: Selected device: -> final score = 1000
SYCL_PI_TRACE[all]:   platform: Intel(R) OpenCL
SYCL_PI_TRACE[all]:   device: Intel(R) Xeon(R) Gold 6354 CPU @ 3.00GHz
```